### PR TITLE
Update comments in Android workflow

### PR DESCRIPTION
**Pull request by mszhanyi**
**Description**: 
Update the comments in Android workflow

**Motivation and Context**
Explain why android CI has to run on Mac agent.

PS.
Move the mac agent info to the PR #11338

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/android-x86_64-crosscompile-ci-pipeline.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Known Limits
 # 1. Anchors are not supported in GHA
 # https://github.community/t/support-for-yaml-anchors/16128/90
-# 2. Nested Virutalizaiton isn't supported in Azure pipeline
-# https://developercommunity.visualstudio.com/t/enable-nested-virtualization-on-azure-pipelines/466384
+# 2. today most cloud-based CI services are still lacking hardware acceleration support from the host VM, 
+# which is the no.1 blocker for running tests on modern Android Emulators (especially on recent API levels) on CI.
```
Comment (skottmckay) [inline tools/ci_build/github/azure-pipelines/android-x86_64-crosscompile-ci-pipeline.yml:6]: Is there a source for this info?

**Comment by skottmckay** [comment]
I'm a little confused about this CI. 

It seems to be for Android/NNAPI, and android has linux and windows emulators. Why does any of it need to run on macos? 

I would have thought we'd only use the macos pool to test a mac specific build or if the CoreML EP was involved, but I couldn't see anything that indicated that was the case.

File: tools/ci_build/github/azure-pipelines/android-x86_64-crosscompile-ci-pipeline.yml (hunk 1 above)
Comment (mszhanyi) [reply tools/ci_build/github/azure-pipelines/android-x86_64-crosscompile-ci-pipeline.yml:6]: https://ychescale9.medium.com/running-android-instrumented-tests-on-ci-from-bitrise-io-to-github-actions-a611e57ce71f

**Comment by mszhanyi** [comment]
> I'm a little confused about this CI.
> 
> It seems to be for Android/NNAPI, and android has linux and windows emulators. Why does any of it need to run on macos?
> 
> I would have thought we'd only use the macos pool to test a mac specific build or if the CoreML EP was involved, but I couldn't see anything that indicated that was the case.

I think the mac agent provided by Azure is not container based. So it could enable hardware acceleration support.

File: tools/ci_build/github/azure-pipelines/android-x86_64-crosscompile-ci-pipeline.yml (hunk 1 above)
Comment (mszhanyi) [reply tools/ci_build/github/azure-pipelines/android-x86_64-crosscompile-ci-pipeline.yml:6]: https://github.com/ReactiveCircus/android-emulator-runner/
Pytorch uses it too.

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/android-x86_64-crosscompile-ci-pipeline.yml:6]: @[satyajandhyala](https://github.com/satyajandhyala) told me he successfully ran the emulator in Azure, and it worked fine.  Satya, would you mind sharing more details?

Comment (mszhanyi) [reply tools/ci_build/github/azure-pipelines/android-x86_64-crosscompile-ci-pipeline.yml:6]: @snnn @satyajandhyala any more comments

**Review by snnn — APPROVED** [review]

**Comment by SatyaAtMS** [comment]
I have created an Android build pipeline for ARM64. Please take a look at
https://aiinfra.visualstudio.com/ORT-QNN-ep/_git/ORT-QNN-ep/pullrequest/7759?_a=files
Let me know if you have any questions

**Comment by snnn** [comment]
> Why does any of it need to run on macos?

Because our build machines are virtual machines. Sometimes it is impossible to run another VM in a VM. 

But it looks like @[SatyaAtMS](https://github.com/SatyaAtMS) has resolved it.  Some Azure VMs support nest-virtualization, some of them do not. The ones we use for running our Linux build pipelines are capable of.

**Comment by snnn** [comment]
> I would have thought we'd only use the macos pool to test a mac specific build or if the CoreML EP was involved,

We should. But since the pipelines were setup when Yi hadn't joined the team, it's not the case.

**Pull request merged**